connectd/tor_autoservice.c

```diff
@@ -263,3 +333,45 @@ struct wireaddr *tor_autoservice(const tal_t *ctx,
 
 	return onion;
 }
+
+
+struct wireaddr *tor_fixed_service(const tal_t *ctx,
+				 const struct wireaddr_internal *tor_serviceaddr,
+				 const char *tor_password,
+				 const char *blob,
+				 const struct wireaddr *bind,
+				 const u8 index)
+{
+	int fd;
+	const struct wireaddr *laddr;
+	struct wireaddr *onion;
+	struct addrinfo *ai_tor;
+	struct rbuf rbuf;
+	char *buffer;
+
+	laddr = bind;
+	ai_tor = wireaddr_to_addrinfo(tmpctx, &tor_serviceaddr->u.torservice.torservice_address);
+
+	fd = socket(ai_tor->ai_family, SOCK_STREAM, 0);
+	if (fd < 0)
+		err(1, "Creating stream socket for Tor");
+
+	if (connect(fd, ai_tor->ai_addr, ai_tor->ai_addrlen) != 0)
+		err(1, "Connecting stream socket to Tor service");
+
+	buffer = tal_arr(tmpctx, char, rbuf_good_size(fd));
+	rbuf_init(&rbuf, fd, buffer, tal_count(buffer), buf_resize);
+
+	negotiate_auth(&rbuf, tor_password);
+
+	onion = make_fixed_onion(ctx, &rbuf, laddr, blob, tor_serviceaddr->u.torservice.port);
+	memset((void *)blob, 0, sizeof(*blob));
```

Comment:
Don't destroy this here, it's const so we promised we wouldn't!  If we really want to do this, we need to do it at a higher level.

Reply:
done, clear data moved to connectd.c